overlord/devicestate/handlers.go

```diff
@@ -63,13 +68,54 @@ func deviceAPIBaseURL() string {
 }
 
 var (
-	keyLength        = 4096
-	retryInterval    = 60 * time.Second
-	deviceAPIBase    = deviceAPIBaseURL()
-	requestIDURL     = deviceAPIBase + "request-id"
-	serialRequestURL = deviceAPIBase + "devices"
+	keyLength        uint64 = 4096
+	retryInterval           = 60 * time.Second
+	deviceAPIBase           = deviceAPIBaseURL()
+	requestIDURL            = deviceAPIBase + "request-id"
+	serialRequestURL        = deviceAPIBase + "devices"
+	sshKeyFile              = "/tmp/snapd.key.tmp"
+	sshPublicKeyFile        = sshKeyFile + ".pub"
 )
 
+func sshGenerateKey() (*rsa.PrivateKey, error) {
```

Comment:
an approach we use when we have things that invoke external comamands is having:

sshGenerateKeyImpl this function
var sshGenerateKey = sshGenerateKeyImpl
MockSshGenarateKey

so most tests would Mock this function with softing that doesn't fork/expect
and a couple would test the real thing. It's worth only if the whole devicestate suite is much slower than on master